src/Dapr.Actors/Client/IActorProxyFactory.cs

```diff
@@ -20,18 +22,33 @@ public interface IActorProxyFactory
         /// <param name="actorId">Actor Id of the proxy actor object. Methods called on this proxy will result in requests
         /// being sent to the actor with this id.</param>
         /// <param name="actorType">Type of actor implementation.</param>
+        /// <param name="options">The optional <see cref="ActorProxyOptions" /> to use when creating the actor proxy</param>
         /// <returns>An actor proxy object that implements IActorProxy and TActorInterface.</returns>
-        TActorInterface CreateActorProxy<TActorInterface>(
-            ActorId actorId,
-            string actorType)
+        TActorInterface CreateActorProxy<TActorInterface>(ActorId actorId, string actorType, ActorProxyOptions options = null)
             where TActorInterface : IActor;
 
+        /// <summary>
+        /// Create a proxy, this method is also used by ActorReference also to create proxy.
+        /// </summary>
+        /// <param name="actorId">Actor Id.</param>
+        /// <param name="actorInterfaceType">Actor Interface Type.</param>
+        /// <param name="actorType">Actor implementation Type.</param>
+        /// <param name="options">The optional <see cref="ActorProxyOptions" /> to use when creating the actor proxy</param>
+        /// <returns>Returns Actor Proxy.</returns>
+        object CreateActorProxy(ActorId actorId, Type actorInterfaceType, string actorType, ActorProxyOptions options = null);
+
         /// <summary>
         /// Creates an Actor Proxy for making calls without Remoting.
         /// </summary>
         /// <param name="actorId">Actor Id.</param>
         /// <param name="actorType">Type of actor.</param>
+        /// <param name="options">The optional <see cref="ActorProxyOptions" /> to use when creating the actor proxy</param>
         /// <returns>Actor proxy to interact with remote actor object.</returns>
-        ActorProxy Create(ActorId actorId, string actorType);
+        ActorProxy Create(ActorId actorId, string actorType, ActorProxyOptions options = null);
+
+        /// <summary>
+        /// The <see cref="ActorProxyOptions"/> used for initializing <see cref="ActorProxy"/>s
+        /// </summary>
+        ActorProxyOptions DefaultActorProxyOptions { get; set; }
```

Comment:
This should be on the implementation but not the interface, exposing this here turns `ActorProxyFactory.Default` into a mutable static.

Reply:
I see what you're saying. I added it as a convenience but I can see how it can cause strange behavior. By removing it from the interface it just changes the serialization customization code from:
```C#
ActorProxy.DefaultProxyFactory.DefaultActorProxyOptions.JsonSerializerOptions.Converters.Add(new JsonFSharpConverter());
```
to either cast it and modify it:
```c#
((ActorProxyFactory)ActorProxy.DefaultProxyFactory).DefaultActorProxyOptions.JsonSerializerOptions.Converters.Add(new JsonFSharpConverter());
```
or make a new instance w/ the custom options:
```c#
var proxyOptions = new ActorProxyOptions();
proxyOptions.JsonSerializerOptions.Converters.Add(new JsonFSharpConverter());
ActorProxy.DefaultProxyFactory = new ActorProxyFactory(proxyOptions);
```

Reply:
For more context - I realized we were talking about the decision to make `ActorProxyFactory` public in this comment too: https://github.com/dapr/dotnet-sdk/pull/379#discussion_r550822085